### ISSUE TITLE
Error returned calling AddRecordToActionList / How to dump XML request ?  

### ISSUE THREAD

**Issue by kaizokou** [mediapart/selligent#28]
Hello, 

When I perform a call to the AddRecordToActionList endpoint defined a follow 

```
<s:element name="AddRecordToActionList">
<s:complexType>
<s:sequence>
<s:element minOccurs="1" maxOccurs="1" name="actionListID" type="s:int"/>
<s:element minOccurs="0" maxOccurs="1" name="data" type="tns:ArrayOfProperty"/>
</s:sequence>
</s:complexType>
</s:element>
<s:element name="AddRecordToActionListResponse">
<s:complexType>
<s:sequence>
<s:element minOccurs="1" maxOccurs="1" name="AddRecordToActionListResult" type="s:int"/>
<s:element minOccurs="0" maxOccurs="1" name="errorStr" type="s:string"/>
</s:sequence>
</s:complexType>
</s:element>
```

I received this error from Selligent 
`
{ ["AddRecordToActionListResult"]=> int(10006) ["errorStr"]=> string(34) "Unable to add record to actionlist" }`

To me (the dev) it seems that there is a configuration problem with the list. But the testing team said that it's working with the same parameters in SOAP UI so it's not a list configuration problem but a dev issue ...

==> I don't have access to Selligent backend account. 
==> The team in charge of the Selligent account have contacted them to ask if there is anything wrong with our request / configuration. Here the Selligent support answer. 

> Seeing as the individual API doesn't accept JSON I take it you're using some kind of library to transform these to XML?
> Would it be possible to capture the XML request you're sending somehow?

So here is my question, Is there a way to dump the SOAP request ? Any other suggestion on what might be the problem

Thx.

**Comment by kaizokou** [comment]
hi @methylbro  @Talamasca  any idea how I can dump the XML request ? 
Thx.

**Comment by methylbro** [comment]
Hi @kaizokou :)

[PHP SOAP Client](http://php.net/manual/en/class.soapclient.php) (that we used here) has [a debug mode](http://php.net/manual/en/soapclient.soapclient.php#112970). You could activate it by passing "trace" option to the constructor. With this option, the SOAP Client will preserve each XML Request so you can read them with `$client->__getLastRequest()` for example.

It's something I didn't anticipate at the beginning but I think we should add it now. (I missed it too during my work ;) )
I don't have the time to develop that feature. So don't hesitate to propose a PR :)

**Comment by kaizokou** [comment]
@methylbro 

Thank you for your help. Thanks to you I managed to moved forward. 
To activate the debug mode it's not so obvious so here is how I did 

I put these two lines in the open() function of Mediapart\Selligent\Connection class. 

```
	    $configuration['options']['trace'] = 1;
	    $configuration['options']['exceptions'] = 0;
```

Now when you call __getLastRequest() on your SoapClient instance to see the result in the browser use the htmlentities() function otherwise you will have a weird string result.

`var_dump(htmlentities($this->client->__getLastRequest()));`
